src/components/UI/atoms/Modal/Modal.tsx

```diff
@@ -0,0 +1,46 @@
+import { useRef, useEffect, useState, ReactNode } from "react";
+import { createPortal } from "react-dom";
+import styled from "@emotion/styled";
+
+interface TModal {
+  children: ReactNode;
+}
+const Modal = (props: TModal) => {
+  const ref = useRef<Element | null>(null);
+  const [mounted, setMounted] = useState(false);
+
+  useEffect(() => {
+    ref.current = document.querySelector<HTMLLIElement>("#portal");
+    setMounted(true);
+  }, []);
+  return mounted && ref.current
+    ? createPortal(
+        <Overlay>
+          <ModalContainer>{props.children}</ModalContainer>
+        </Overlay>,
+        ref.current,
+      )
+    : null;
+};
```

Comment:
포탈을 쓴 이유가 뭔지 궁금해요. 사용하지 않고 _app 파일에 컴포넌트 정의하면되지 않나 싶어서요.

그리고 요렇게 한줄에 많은 조건을 다는건 별로 안좋은 것 같아요 줄은 늘어나지만 if 문이 나중에 로직과 조건이 복잡해도 확장성과 가독성이 좋아질꺼에요.
```suggestion
  useEffect(() => {
    ref.current = document.querySelector<HTMLLIElement>("#portal");
    setMounted(true);
  }, []);
  
  if (!mounted || !ref.current) {
    return null;
  }
  
  return createPortal(
        <Overlay>
          <ModalContainer>{props.children}</ModalContainer>
        </Overlay>,
        ref.current,
      );
};
```

Reply:
모달을 만들려고 당연하게 portal을 사용하는것을 생각했던 것 같습니다, 찾아보니 _app 파일에서 모달을 생성할 수 있는 것 같은데 뭐를 사용할지 좀 더 고민해보겠습니다!